internal/core/operations/healthcontroller/executor.go

```diff
@@ -451,6 +452,32 @@ func GetDesiredNumberOfRooms(
 	return
 }
 
+func AdjustDownscaleAmount(ctx context.Context, availableRooms []string, amountToBeDeleted int, roomStorage ports.RoomStorage, scheduler *entities.Scheduler, logger *zap.Logger) (downscaleAmount int) {
```

Comment:
wdyt of naming this something like `CapRollingUpdateDownscaleAmount`

Reply:
Nice, changed